scrc/dataset_creation/criticality_dataset_creator.py

```diff
@@ -1,93 +1,359 @@
-#  People go to 2nd instance much more often than to 3rd instance.
-#  In many cases more than 50% of cases go to 2nd instance.
-#  Probably less than 30% of cases go to 3rd instance. ==> Ask Daniel Kettiger
-# TODO calculate this: Geschäftsberichte von grossen Gerichten holen und vergleichen mit Anzahl Bundesgerichten
 
 class CriticalityDatasetCreator(DatasetCreator):
     """
-    Creates a dataset with the text as input and whether it reaches the supreme court or not as labels
+    Creates a dataset containing bger cases and sets for each case a criticality label, based if the case was published
+    as bge or not.
     """
 
     def __init__(self, config: dict):
         super().__init__(config)
         self.logger = get_logger(__name__)
-
         self.debug = False
         self.split_type = "date-stratified"
         self.dataset_name = "criticality_prediction"
-        # TODO wait for section splitting in other courts for facts and considerations to be enabled
-        self.feature_cols = ['full_text']  # ['facts', 'considerations', 'text']
+        self.feature_cols = ['facts', 'considerations']
+        self.load_rulings()
+        self.extract_bge_references()
+        self.CITATION_COUNT_MIN = 1
 
-    def get_dataset(self, feature_col, lang, save_reports):
-        engine = self.get_engine(self.db_scrc)
+    def get_dataset(self, feature_col, save_reports):
+        """
+        get of all bger cases and set bge_label and citation_label
+        :param feature_col:     specify sections to add as column
+        :param save_reports:    whether or not to compute and save reports
+        :return:                dataframe and list of labels
+        """
+        df = self.get_df(self.get_engine(self.db_scrc), feature_col)
+        self.logger.info(f"There are a total of {len(df.index)} bger cases.")
+        bge_list = self.get_bge_criticality_list()
+        df = self.set_criticality_label(df, bge_list, 'bge_label')
+        criticality_list = self.get_citations_criticality_list(df)
+        df = self.set_criticality_label(df, criticality_list, 'criticality_label')
+        df = self.filter_cases(df)
+        bge_labels, _ = list(np.unique(np.hstack(df.bge_label), return_index=True))
+        return df, bge_labels
 
-        origin_chambers, supreme_court_df = self.query_supreme_court(engine, lang)
+    def filter_cases(self, df):
+        """
+        Filters given dataframe, to get rid of not needed data
+        :param df:      dataframe of all bger cases
+        :return:        filtered dataframe
+        """
+        # TODO change drop to column names in case df columns get's back in a different order
+        #  df.drop('decision_id, axis=1) does not work why?
+        df.drop(df.iloc[:, 15:20], inplace=True, axis=1)
+        df.drop(df.iloc[:, 8:14], inplace=True, axis=1)
+        df.drop(df.iloc[:, 1:6], inplace=True, axis=1)
```

Comment:
what does this do? Do you drop columns?
I would do it like this: df.drop(columns=['B', 'C'])
You have the advantage that it is more readable and also more robust. What if the order of the columns in the df change for some reason?

Reply:
Solved now with column names. Probelm was to set param inplace=True.

Reply:
great